resolver/src/main/java/io/netty/resolver/InetNameResolver.java

```diff
@@ -40,32 +37,15 @@
      */
     protected InetNameResolver(EventExecutor executor) {
         super(executor);
-        loopbackAddress = SocketUtils.loopbackAddress();
     }
 
     /**
+     * @deprecated Use {@link SocketUtils#loopbackAddress()}.
      * Returns the {@link InetAddress} for loopback.
      */
+    @Deprecated
     protected InetAddress loopbackAddress() {
-        return loopbackAddress;
-    }
-
-    @Override
-    public Future<InetAddress> resolve(String inetHost, Promise<InetAddress> promise) {
```

Comment:
Yes this did come up https://github.com/netty/netty/pull/6258#discussion_r97187696 but I'm not convinced this adds value as we will have to check for it at the lower layer anyways. My new train of though is to keep the scope limited to where we actually need to enforce the constraint instead of having it spread out in different places. The spread out nature was one of the reasons why this bug exists in the first place.